### ISSUE TITLE
Add note in connections.jade about hostname and `members.host` from `rs.conf()`

### ISSUE THREAD

**Issue by haydursek** [Automattic/mongoose#7006]
0
down vote
favorite

I am connecting from my nodejs app using mongoose to a MongoDB replica set in AWS. This connection is made via the private IP of the DB instances, and all was working fine.

Recently, I was doing some testing and upgraded to node 8.11.2, and now I notice that when I try to establish the connection via mongoose, somehow it is converting the private IP of the DB to a public IP and connecting that way. However, as per our security that is blocked, and so the connection fails.

I have gone through our network setup with AWS support and everything seems to be fine on the AWS/network side. From the same server, if I try to connect to the DB instance using the private IP but on mongo shell, it works fine, but the mongoose connection fails via the code in node fails. All connectivity between servers is up and good.

I am using node v8.11.2 and mongodb 3.2.

**Comment by vkarpov15** [comment]
Have you tried connecting just using the MongoDB node driver? That would help determine whether this is a mongoose issue or a MongoDB driver issue.

Also, what version of Mongoose are you using?

**Comment by vkarpov15** [comment]
@haydursek I just worked through a similar issue at my day job. Connect using the `mongo` shell and run `rs.conf()`. Make sure that the `members.$.host` property contains the private IP, not the public IP. If the replica set config shows the public IP, the MongoDB driver will use the public IP, even if the private IP is what you specified in the connection string. [This page tells you how to modify the `host` property without any downtime](https://docs.mongodb.com/manual/tutorial/change-hostnames-in-a-replica-set/#change-hostnames-while-maintaining-replica-set-availability)

**Comment by haydursek** [comment]
@vkarpov15 Sorry, I didn't respond earlier, but I found our issue and it was exactly that. To be more specific, we have a strange setup of having the public ip in our rs.conf, but then use the private ip to connect to our instances. So i was able to trace the error that I was getting, which was mongo complains that the ip of the master in the rs.conf does not match the ip with which we are trying to connect, and thus blows up.

There was even an open issue for this exact case, but was closed as expected behavior. I have been able to get around it for now by opening up ports on our firewall, because even though we are using the private ip to connect, mongo will actually eventually connect over the public ip due to our rs.conf.  I am going to change our rs.conf to private ips, although i remember we had that setup was well and then couldnt connect from any public ip.  So I will let you know how it goes. BTW, when i do revert to putting private ips in the rs.conf, i still bind to 0.0.0.0 in the mongod.conf file correct ?
